pwiz_tools/Skyline/TestFunctional/EADZIonsTest.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System.Globalization;
+using System.Linq;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using pwiz.Common.SystemUtil;
+using pwiz.Skyline.Controls.SeqNode;
+using pwiz.Skyline.Model;
+using pwiz.Skyline.Properties;
+using pwiz.Skyline.SettingsUI;
+using pwiz.Skyline.Util;
+using pwiz.SkylineTestUtil;
+
+namespace pwiz.SkylineTestFunctional
+{
+    /// <summary>
+    /// Testing z+1 and z+2 ions feature.
+    /// </summary>
+    [TestClass]
+    public class EADZIonsTest : AbstractFunctionalTestEx
+    {
+        [TestMethod]
+        public void TestEADIons()
+        {
+            Run(@"TestFunctional\EADZIonsTest.zip");
+        }
+
+        protected override void DoTest()
+        {
+            //Making sure the input aliases are unique across all ion types to avoid input ambiguity.
+            Assert.IsTrue(Enumerable.Range(0, (int) IonType.zhh + 1).SelectMany(i => ((IonType) i).GetInputAliases(), (i, s) => s)
+                .GroupBy(s => s, (s, enumerable) => enumerable.Count()).All(i => i == 1));
+
+            OpenDocument("EADZIonsTest.sky");
+            var transitionSettingsUI = ShowDialog<TransitionSettingsUI>(SkylineWindow.ShowTransitionSettingsUI);
+            RunUI(() =>
+            {
+                transitionSettingsUI.FragmentTypes = "c, z., z'";
+                transitionSettingsUI.IonCount = 5;
+            });
+            OkDialog(transitionSettingsUI, transitionSettingsUI.OkDialog);
+            WaitForGraphs();
+            var libMatch = SkylineWindow.GraphSpectrum.DisplayedSpectrum;
+            ImportResults("FilteredScans\\LITV56_EAD" + ExtensionTestContext.ExtMzml);
+            WaitForGraphs();
+            FindNode((505.5810).ToString("F4", LocalizationHelper.CurrentCulture) + "+++");
+            Settings.Default.ShowZHIons = true;

```

Comment:
You can start your test like this, though it should be done inside RunUI(). It is not safe to alter Settings.Default off the UI thread. It is not a thread-safe class. Otherwise, you should be calling the SkylineWindow.ShowZhIons(true) method I suggested where you have added the menu event handlers, also inside a RunUI() call.

Reply:
I replaced the menu event handlers, but setting a property updates UI, AFAIK, it's in the property setter.